File: leetcode/editor/cn/383_赎金信_test.py
import unittest


class TestSolution(unittest.TestCase):

    def setUp(self) -> None:
        self.solution = Solution()

    def test_givenNormal_when_thenSuccess(self):
        self.assertTrue(self.solution.canConstruct('aa', 'aaa'), "ShouldBeTrue")

    def test_givenFailCase1_when_thenSuccess(self):
        self.assertFalse(self.solution.canConstruct('a', 'b'))


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        ans = True
        mapping = {}
        for c in magazine:
            if mapping.get(c) is None:
                mapping[c] = 1
            else:
                mapping[c] += 1
        for c in ransomNote:
            cnt = mapping.get(c)
            if cnt is None or cnt < 1:
                ans = False
                break
            mapping[c] -= 1
        return ans


# leetcode submit region end(Prohibit modification and deletion)


if __name__ == '__main__':
    unittest.main()

File: leetcode/editor/cn/383_赎金信_test.go
package leetcode

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func Test_givenNormal_when383_thenSuccess(t *testing.T) {
	assert.Equal(t, true, canConstruct("aa", "baa"), "shouldEqual")
}

//leetcode submit region begin(Prohibit modification and deletion)
func canConstruct(ransomNote string, magazine string) bool {
	ans := true
	var mapping [26]int
	for _, r := range magazine {
		mapping[r-'a'] += 1
	}
	for _, r := range ransomNote {
		cnt := mapping[r-'a']
		if cnt < 1 {
			ans = false
			break
		}
		mapping[r-'a'] -= 1
	}
	return ans
}

//leetcode submit region end(Prohibit modification and deletion)
